### PR TITLE
Refactor repeat count

### PR DESCRIPTION
**Pull request by JakeLin**
In this PR, we have done some API improvements for Swift 3
1. We remove `repeatCount` from `Animatable` protocol and add it in `AnimationType` enum. 
2. Remove `way` from some animations like `slide`, it reads better like `containerPickerView.slide(.in, direction: .up)`
3. Make `animateIn` and `animateOut` methods more Swifty.
4. Group `private` methods in a private extension.
5. Add `squash` animation for version 3. Same as `squeeze` in version 2.
6. Update the animation playground demo. I really like @lastMove 's demo app, very flexible to add new animations to the demo.

@lastMove can you have a look since you made most of changes for `Animatable` an`AnimationType`

@tbaranes please have a look as well. 

Related to #221

### CODE REVIEW

**Comment by tbaranes** [comment]
👍

File: IBAnimatable/Animatable.swift [hunk 1]
```diff
@@ -42,11 +42,6 @@ public protocol Animatable: class {
    Animation force (default value should be 1)
    */
   var force: CGFloat { get set }
-
```
Comment (tbaranes) [inline IBAnimatable/Animatable.swift]: kapok

File: IBAnimatable/Animatable.swift [hunk 2]
```diff
@@ -42,11 +42,6 @@ public protocol Animatable: class {
    Animation force (default value should be 1)
    */
   var force: CGFloat { get set }
-  
-  /**
-   Repeat count for Shake, Pop, Morph, Squeeze, Flash, Wobble and Swing animations
-   */
-  var repeatCount: Float { get set }
```
Comment (tbaranes) [inline IBAnimatable/Animatable.swift]: Testing new github review stuff

**Review by tbaranes — APPROVED** [review]
Seems good 👍

**Comment by lastMove** [comment]
"Remove repeatCount from Animatable protocol and add it in AnimationType enum" Great idea !

**Review by lastMove — APPROVED** [review]
All good

**Comment by JakeLin** [comment]
Thanks @tbaranes @lastMove

**Pull request merged**